spec/rspec/mocks/nil_expectation_warning_spec.rb

```diff
@@ -15,13 +15,27 @@ module Mocks
         expect(nil).not_to receive(:foo)
       end
 
-      it "does not issue a warning when expectations are set to be allowed" do
-        allow_message_expectations_on_nil
```

Comment:
While the new global config way of configuring this is the way we'll recommend going forward, it's important that we still support this old interface (since it's a public API) until RSpec 4, so we'll need to keep existing specs around for it.

I do think we should soft-deprecate it, though.  Can you update the YARD docs for that method and add a [deprecated](http://www.rubydoc.info/gems/yard/file/docs/Tags.md#deprecated) tag?  We won't issue a deprecation warning yet -- that'll come in 3.99 as part of the upgrade-to-RSpec-4 process -- but the YARD docs will help direct people to the new way to configure this.